File: 351004/Brazhalovich/Lab_1/src/main/java/org/example/newsapi/controller/NewsController.java
package org.example.newsapi.controller;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.example.newsapi.dto.request.NewsRequestTo;
import org.example.newsapi.dto.response.NewsResponseTo;
import org.example.newsapi.service.NewsService;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/news")
@RequiredArgsConstructor
public class NewsController {
    private final NewsService newsService;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public NewsResponseTo create(@RequestBody @Valid NewsRequestTo request) {
        return newsService.create(request);
    }

    @GetMapping
    public List<NewsResponseTo> getAll() {
        return newsService.findAll();
    }

    @GetMapping("/{id}")
    public NewsResponseTo getById(@PathVariable Long id) {
        return newsService.findById(id);
    }

    @PutMapping("/{id}")
    public NewsResponseTo update(@PathVariable Long id, @RequestBody @Valid NewsRequestTo request) {
        return newsService.update(id, request);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void delete(@PathVariable Long id) {
        newsService.delete(id);
    }
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/controller/NewsController.kt
package com.distcomp.controller

import com.distcomp.dto.news.NewsRequestTo
import com.distcomp.dto.news.NewsResponseTo
import com.distcomp.service.NewsService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/{version}/news")
class NewsController (
    private val newsService: NewsService
) {
    @PostMapping(version = "1.0")
    @ResponseStatus(HttpStatus.CREATED)
    fun createUser(@RequestBody newsRequestTo: NewsRequestTo): NewsResponseTo {
        return newsService.createNews(newsRequestTo)
    }

    @GetMapping("{id}")
    fun readUserById(@PathVariable("id") id: Long): NewsResponseTo {
        return newsService.readNewsById(id)
    }

    @GetMapping
    fun findAll(): List<NewsResponseTo> {
        return newsService.readAll()
    }

    @PutMapping(path = ["/{id}", ""], version = "1.0")
    @ResponseStatus(HttpStatus.OK)
    fun updateUser(@RequestBody newsRequestTo: NewsRequestTo,
                   @PathVariable("id") userId: Long?): NewsResponseTo {
        return newsService.updateNews(newsRequestTo, userId)
    }

    @DeleteMapping(path = ["/{id}"], version = "1.0")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteUser(@PathVariable("id") id: Long) {
        newsService.removeNewsById(id)
    }
}